branchVal.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+local_branch=${1}
+
+valid_branch_dash="([a-z0-9\-\]+)"
+
+
+message="There is something wrong with your branch name. Branch names in this project must adhere to this contract: $valid_branch_regex, $local_branch. Your commit will be rejected. You should rename your branch to a valid name and try again."
```

Comment:
I'm very biased here, since I wrote the original message.  But I prefer the older style of the error message.
See:  https://github.com/CMSgov/macpro-quickstart-serverless/blob/master/.github/workflows/deploy.yml#L25-L40
I like that it spells out that this is a serverless constraint, and also tries to explain in plain english the constraints.
It would have to be modified to explain that upper cases letters are no good.
Consider going back to that more verbose message format.

Reply:
agreed with this ^^